mpl_interactions/controller.py

```diff
@@ -291,6 +293,16 @@ def save_animation(
             else:
                 N = int((max_ - min_) / slider.valstep)
                 step = slider.valstep
+        elif not _not_ipython and isinstance(slider, (FloatSlider, IntSlider)):
+            ipywidgets_slider = True
+            min_ = slider.min
+            max_ = slider.max
+            step = slider.step
+            N = int((max_ - min_) / step)
+        else:
+            raise NotImplementedError(
```

Comment:
The `UnboundLocalError` can be fished out with pylint. Maybe an idea to lint the codebase with that?

Reply:
Yup makes sense to me. But I just tried running `pylint` and oh boyyyyy there are many issues it raises, so let's not do that here

Reply:
That's right, this is probably a longer project :S Another option is to add a pre-commit hook and address pylint issues step by step once files are modified. It's also possible to ignore files or issues through `.pylintrc`, or through inline ignores.

Reply:
flake8 is also an option to consider, as well as some of its plugins